src/ZkFold/Symbolic/Algorithms/Hash/SHA2.hs

```diff
@@ -186,56 +204,66 @@ type SHA2 algorithm context k =
 -- >>> bs = fromConstant (42 :: Natural) :: ByteString 8 (Zp BLS12_381_Scalar)
 -- >>> hash = sha2 @"SHA256" bs
 --
+
+divisibleDiv :: forall a b. (Mod a b ~ 0) :- (Div a b * b ~ a)
+divisibleDiv = Sub unsafeAxiom
+
+withDivisibleDiv :: forall a b {r}. Mod a b ~ 0 => ((Div a b * b ~ a) => r) -> r
+withDivisibleDiv = withDict (divisibleDiv @a @b)
+
 sha2
     :: forall (algorithm :: Symbol) context k
     .  SHA2 algorithm context k
+    => Div (PaddedLength k (ChunkSize algorithm) (2 * WordSize algorithm)) (ChunkSize algorithm) * (ChunkSize algorithm) ~ PaddedLength k (ChunkSize algorithm) (2 * WordSize algorithm)
```

Comment:
Evidence for these should be cooked up inside `sha2`, as these are semantically true and should not be present in the public interface